### PR TITLE
Add abi skeleton

### PR DESCRIPTION
**Pull request by AgwaB**
details:
- abi
- method
- argument
- type

we need to define argument type which will used in koa

### CODE REVIEW

File: abi/abi.go [hunk 1]
```diff
@@ -16,22 +16,31 @@
...
-// The implementation below is implemented for the abi spec of ethereum.
-// https://solidity.readthedocs.io/en/develop/abi-spec.html
+type ABI struct {
+	Methods []Method
+}
 
-// Encode abi parameters
-func Encode(params ...[]byte) []byte {
-	return nil
+func New(abiJSON string) (ABI, error) {
```
Comment (boohyunsik) [inline abi/abi.go:30]: Why don't you return *ABI?

File: abi/abi.go [hunk 2]
```diff
@@ -16,22 +16,31 @@
...
+	reader := strings.NewReader(abiJSON)
+	dec := json.NewDecoder(reader)
+
+	var abi ABI
+	if err := dec.Decode(&abi); err != nil {
+		return ABI{}, err
```
Comment (boohyunsik) [inline abi/abi.go:36]: Can you return nil instead of ABI? (It is about return type)

File: abi/abi.go (hunk 1 above)
Comment (AgwaB) [reply abi/abi.go:30]: ABI is just value and there is no change in ABI!

File: abi/abi.go (hunk 2 above)
Comment (AgwaB) [reply abi/abi.go:36]: ok! there is no reason to return `ABI{}` instead of `nil` ... :)

File: abi/abi.go [hunk 3]
```diff
@@ -16,22 +16,31 @@
...
+// The implementation below is implemented for the abi spec of ethereum.
+// https://solidity.readthedocs.io/en/develop/abi-spec.html
+
 // Encode abi parameters with function Selector
 func EncodeWithSelector(selector []byte, params ...[]byte) []byte {
 	return nil
 }
-
-// Get function selector(4bytes) from string of function signature
-func Selector(functionSignature string) []byte {
```
Comment (zeroFruit) [inline abi/abi.go]: Why did you removed this function? just my curiosity

**Review by junbeomlee — APPROVED** [review]

**Pull request merged**